hack/postrelease/tests/test_manifests_archives.py

```diff
@@ -24,31 +24,37 @@
     "custom-resources.yaml",
 ]
 
-
-def setup_archive():
-    response = requests.get(url, stream=True)
-    global file
-    file = tarfile.open(fileobj=response.raw, mode="r|gz")
-
-
-def teardown_archive():
-    file.close()
-
-
-@with_setup(setup=setup_archive, teardown=teardown_archive)
-def test_manifest_present():
-    for manifest in manifest_list:
-        yield check_manifest_present, manifest
-
-
-def check_manifest_present(manifest):
-    print("[INFO] checking {} is in archive".format(manifest))
+OVERRIDE_FILE = os.getenv("OVERRIDE_FILE")
+
+
+@pytest.fixture(scope="session")
+def image_file_members(tmpdir_factory):
+    if OVERRIDE_FILE:
+        tf = tarfile.open(OVERRIDE_FILE, "r|gz")
```

Comment:
I think you'll want to use a context manager to keep track of the opened file so that it will be closed automatically when the context ends. That or call `close()` manually.

Reply:
Done!